### PR TITLE
#167123375 Users be able to login via social auth

### PR DESCRIPTION
**Pull request by kelvinndmo**
## Description ##
- Enables users to be to login to the application via social authentication.

## Type of change ##
Please select the relevant option
- [ ] Bug fix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [ ] Integration test
- [x] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[#167123375](https://www.pivotaltracker.com/story/show/167123375)

### CODE REVIEW

**Comment by malep2007** [comment]
@kelvinndmo, please rebase and get back to me on this

File: src/environments/environment.ts [hunk 1]
```diff
@@ -3,7 +3,14 @@
 // The list of file replacements can be found in `angular.json`.
 
 export const environment = {
-  production: false
+  production: false,
+  facebookId: '1782833315349267',
+  googleId:
+    '315082406739-imasn7gsabthac3uaer84oqmp1g2780p.apps.googleusercontent.com',
+  googleUrl:
```
Comment (akram256) [inline src/environments/environment.ts]: Hey @kelvinndmo, we have a base-url already created by Brain, Why don't we just use that.What do u think?

**Review by akram256 — CHANGES_REQUESTED** [review]

File: src/environments/environment.ts (hunk 1 above)
Comment (kelvinndmo) [reply src/environments/environment.ts]: sure

File: src/app/pages/SocialAuth/socialauth.component.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  selector: 'app-social-login',
+  templateUrl: './socialauth.component.html',
+  styleUrls: ['./socialauth.component.scss']
+})
+export class SocialLoginComponentt {
+  user: SocialUser = null;
+  loggedIn: boolean;
+  loading: boolean = true;
+
+  // inject services and depe
```
Comment (MandelaK) [inline src/app/pages/SocialAuth/socialauth.component.ts:23]: This comment is not really helpful, could you please do away with it or make it more helpfule?

File: src/app/pages/login/login-form/login-form.component.spec.ts [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
 import { LoginSliderComponent } from '../login-slider/login-slider.component';
-import { SocialLoginComponent } from '../social-login/social-login.component';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { LoginService } from 'src/app/services/login/login.service';
-import { loginServiceSpy, resetSpies, toastServiceSpy } from '../../../helpers/spies';
+import {
+  loginServiceSpy,
+  resetSpies,
+  toastServiceSpy
+} from '../../../helpers/spies';
```
Comment (MandelaK) [inline src/app/pages/login/login-form/login-form.component.spec.ts:17]: Please us absolute imports

Comment (kelvinndmo) [reply src/app/pages/login/login-form/login-form.component.spec.ts:17]: this seems to have been merged with s previous PR

**Review by MandelaK — APPROVED** [review]
Everything works okay, good job

File: src/app/config.ts [hunk 4]
```diff
@@ -1,15 +1,17 @@
-function appConfig () {
-   
-    return {
-        base_url: 'http://127.0.0.1:8000/api/v1/'
-    };
+function appConfig() {
+  return {
+    base_url: 'https://landville-backend-web-api.herokuapp.com/api/v1/',
+    facebookId: '1782833315349267',
+    googleId:
```
Comment (CryceTruly) [inline src/app/config.ts:7]: @kelvinndmo, Could there be any catch including the google id and the facebook app id in version control?
If that's the case then we could consider using environment variables.

File: src/app/pages/SocialAuth/socialauth.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    spyOn(component, 'signInWithGoogle').and.callThrough();
+    let button = fixture.debugElement.query(By.css('.google'));
+
+    button.triggerEventHandler('click', null);
+    fixture.detectChanges();
+    expect(component.signInWithGoogle).toHaveBeenCalled();
+  });
+
+  it('test if sign with facebook', () => {
```
Comment (CryceTruly) [inline src/app/pages/SocialAuth/socialauth.component.spec.ts]: @kelvinndmo, it's customary that 'it' is followed by should, We should also try and make the test suit descriptions as intuitive as possible.

**Review by CryceTruly — CHANGES_REQUESTED** [review]

**Review by Genza999 — COMMENTED** [review]
Nice work @kelvinndmo , LGTM

**Review by akram256 — APPROVED** [review]
LGTM

**Review by Genza999 — APPROVED** [review]

**Review by BrianSerem — APPROVED** [review]

**Review by JamesMudidi — CHANGES_REQUESTED** [review]
@kelvinndmo, Good job you have done here. Just to add on what @CryceTruly had already said, it would be better to have our app ids in a .env file. here are some resources to help you get started with that,

1. https://blog.usejournal.com/system-environment-variables-in-angular-1f4a922c7b4c
2. https://www.jvandemo.com/how-to-use-environment-variables-to-configure-your-angular-application-without-a-rebuild/
3. https://medium.com/beautiful-angular/angular-2-and-environment-variables-59c57ba643be

**Review by joelethan — APPROVED** [review]

**Review by CryceTruly — APPROVED** [review]

**Review by JamesMudidi — CHANGES_REQUESTED** [review]
LGTM

**Review by JamesMudidi — APPROVED** [review]
LGTM

**Review by Victoradukwu — APPROVED** [review]

**Comment by malep2007** [comment]
@kelvinndmo CicleCI fails

**Pull request merged**